### PR TITLE
Implement a basic Shopify input driver

### PR DESCRIPTION
**Pull request by emk**
This uses the Shopify REST API and an accessToken to download JSON
data and convert it to CSV with embedded structs. This can be used as
follows:

    dbcrossbar cp \
      --if-exists=upsert-on:id \
      --temporary=gs://$TEMP_BUCKET \
      --temporary=bigquery:$PROJECT:$TEMP_DATASET \
      --schema='dbcrossbar-ts:shopify.ts#Order' \
      shopify://$SHOP/admin/api/2020-04/orders.json?status=any \
      bigquery:$PROJECT:$DATASET:$DEST_TABLE

This patch also includes a fix for `null` values in JSON where
optional arrays or structs are expected, which occurs frequently with
the Shopify API. And we update `shopify.ts` to make most fields
nullable, with the exception of `id` fields (which are useful for
upserts).

### CODE REVIEW

File: dbcrossbarlib/src/drivers/shopify/mod.rs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            .parse::<Url>()
+            .with_context(|_| format_err!("could not set URL scheme for {}", self))?;
+        Ok(https_url)
+    }
+}
+
+#[test]
+fn to_https_url() {
+    let loc = ShopifyLocator::from_str(
+        "shopify://example.myshopify.com/admin/api/2020-04/orders.json",
```
Comment (seamusabshere) [inline dbcrossbarlib/src/drivers/shopify/mod.rs:36]: ah FQDN not just shop name

File: dbcrossbarlib/src/drivers/shopify/local_data.rs [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+                sender.send(Err(err)).await.map_send_err()?;
+                return Ok(());
+            }
+            include_headers = false;
+
+            // Figure out what to do next.
+            if let Some(next_page_url) = resp.next_page_url {
+                next_url = next_page_url;
+
+                // If we're starting to overheat, wait a full second, giving enough
```
Comment (seamusabshere) [inline dbcrossbarlib/src/drivers/shopify/local_data.rs:87]: we'll have to see how well this flows

File: dbcrossbarlib/src/drivers/shopify/mod.rs (hunk 1 above)
Comment (emk) [reply dbcrossbarlib/src/drivers/shopify/mod.rs:36]: Shop names are always given as FQDNs by the Shopify API, and I don't dare to tamper with that.

File: dbcrossbarlib/src/drivers/shopify/local_data.rs (hunk 2 above)
Comment (emk) [reply dbcrossbarlib/src/drivers/shopify/local_data.rs:87]: On average, we replenish 2 tokens/sec (or 4 tokens/second). There's also a burst pool. If we've used up less than half the burst pool, we run once a second and use as many tokens as we can until the burst pool is half full again.

**Pull request merged**